cpp/CMakeLists.txt

```diff
@@ -55,6 +55,7 @@ option(BUILD_SHARED_LIBS "Build cuvs shared libraries" ON)
 option(BUILD_TESTS "Build cuvs unit-tests" ON)
 option(BUILD_C_LIBRARY "Build raft C API library" OFF)
 option(BUILD_C_TESTS "Build raft C API tests" OFF)
+option(BUILD_ANN_BENCH "Build cuVS ann benchmarks" OFF)
```

Comment:
Can we add an option to build this in CI? 

It looks to me that this is off by default (which is fine), but that does mean that we don't have any guarantee that any of this is compiling.

Reply:
We enable this currently together with tests in build.sh (see https://github.com/rapidsai/cuvs/pull/130#discussion_r1642994596)